app/dashboard/views/mailbox_detail.py

```diff
@@ -179,12 +179,53 @@ def mailbox_detail_route(mailbox_id):
 
         elif request.form.get("form-name") == "toggle-pgp":
             if request.form.get("pgp-enabled") == "on":
+                if not mailbox.disable_smime:
+                    mailbox.disable_smime = True
+                    flash(f"S/MIME is disabled on {mailbox.email}", "warning")
                 mailbox.disable_pgp = False
                 flash(f"PGP is enabled on {mailbox.email}", "success")
             else:
                 mailbox.disable_pgp = True
                 flash(f"PGP is disabled on {mailbox.email}", "info")
 
+            Session.commit()
+            return redirect(
+                url_for("dashboard.mailbox_detail_route", mailbox_id=mailbox_id)
+            )
+        elif request.form.get("form-name") == "smime":
+            if request.form.get("action") == "save":
+                if not current_user.is_premium():
+                    flash("Only premium plan can add S/MIME Key", "warning")
+                    return redirect(
+                        url_for("dashboard.mailbox_detail_route", mailbox_id=mailbox_id)
+                    )
+
+                mailbox.smime_public_key = request.form.get("smime")
```

Comment:
Can you:
 - Verify that this is a valid PEM file.
 - Add a check that the key is not bigger than 15KiB? A PEM chain with several certs shouldn't be bigger than that.